### DISCUSSION TITLE
Support .xyz/.rgb + swizzling in `ti.Vector`

### DISCUSSION THREAD

**Discussion by k-ye** [taichi-dev/taichi#4047] [Ideas]
I think we can specialize up to 4-dimensional vectors by supporting these features:

* .uv, xyz, rgba: These are fields with physical meanings.
* vector swizzle: https://en.wikipedia.org/wiki/Swizzling_(computer_graphics). Another feature commonly used in GLSL

cc @bobcao3 @neozhaoliang @strongoier

**Comment by neozhaoliang** [comment]
Fully agree. Emancipating the productive forces for graphics. :laughing: 

Another important feature in GLSL is the initialization by concatenating low dimension vectors, e.g. `vec4(vec2, vec2)`. Could be tricky to do so in python.

+ `vec2`: `uv`, `xy`
+ `vec3`: `xyz`, `rgb`
+ `vec4`: `xyzw`, `rgba`

**Comment by k-ye** [comment]
Done in https://github.com/taichi-dev/taichi/issues/4810
